### PR TITLE
Fixes #9290: Add an API endpoint to call rudder-remote-run

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9290

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala [hunk 1]
```diff
@@ -81,4 +90,66 @@ class NodeApiService8 (
...
+      while (read != -1) {
+        Arrays.fill(bytes,zero)
+        read = in.read(bytes)
+        out.write(bytes)
+        out.flush()
+      }
+
+    } catch {
+      case e : IOException =>
+        // should we log ? Should we end ?
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala]: You can't do that, or you will leak ressource. The best way to handle that is to let the exception throws here, and manages it above. As your method is only used locally, it's ok. So remove try/catch here.

File: rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala [hunk 2]
```diff
@@ -81,4 +90,66 @@ class NodeApiService8 (
...
+        case Nil => Nil
+        case classes => "-D" :: classes.mkString(",") :: Nil
+      }
+      import scala.collection.JavaConversions._
+      val pb = new NuProcessBuilder("/usr/bin/rudder" :: action ::: classOption)
+      val handler = new RunHandler(out)
+      pb.setProcessListener(handler)
+      pb.start();
+
+      runResponse(in)
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala]: Here, add a try catch, and on exception, close all the open resources: in, out

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**